### PR TITLE
Cleanup the Hyper-V cleanup

### PR DESCRIPTION
**Pull request by vuntz**
I don't think it makes sense to remove half of the code and half of the files. So instead, restore small bits, add an exception to the code in case the Hyper-V setup is done and would fail due to the large files missing, and properly remove the OS from the list of supported OS, with a migration.

### CODE REVIEW

**Comment by susecloudcommitpolice** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please limit commit subject line to 50 characters.<br />
e4a0b59dcfffd306a021b663a5c4d156999235cf</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please limit commit subject line to 50 characters.<br />
123401a54fc887ceff6ee34c2e84ea23865ecd29</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please limit commit subject line to 50 characters.<br />
1c56599bfb77ba2878070d4835ef51606be615d0</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by bmwiedemann** [comment]
I see that the migration is useful, but why keep .ps1 files around? YAGNI says, If they are needed again later, they can be added back later.
And having some leftover 'if windows' codepathes does not hurt. Didnt want to spend more time on finding those.

**Comment by vuntz** [comment]
> I see that the migration is useful, but why keep .ps1 files around? YAGNI says, If they are needed again later, they can be added back later.

For the same reason we have templates for RHEL nodes, and also because there's at least one other windows-only template that wasn't removed in the initial PR.

It doesn't hurt to keep them, and the argument in the initial PR was only about the size -- it doesn't apply for them.

File: chef/cookbooks/provisioner/recipes/setup_base_images.rb [hunk 1]
```diff
@@ -342,6 +342,90 @@
   dirty = true
 end
 
+unless node[:provisioner][:supported_oses].keys.select{ |os| /^(hyperv|windows)/ =~ os }.empty?
+  raise "Binary files (chef-client, curl) need to be added back to the cookbook for Hyper-V support"
```
Comment (dirkmueller) [inline chef/cookbooks/provisioner/recipes/setup_base_images.rb]: add a link to the PR/git commit to revert?

**Pull request merged**